corehq/apps/locations/bulk_management.py

```diff
@@ -883,11 +884,24 @@ def append_at_bottom(parent_type):
 
         return top_to_bottom_locations
 
+    def handle_delete(loc):
+        delete_locations.append(loc)
+        while True:
```

Comment:
Why is this in a `while True` loop?  It doesn't look like anything changes between iterations except the contents of `ancestors_of_deleted_nodes`.  Is it missing a `loc = parent_stub` line at the bottom?

Reply:
good catch. `parent_stub = loc.new_parent_stub` was the thing that was supposed to do it, but obviously that doesn't work. Will fix this.